compiler/resolution/implementForallIntents.cpp

```diff
@@ -1778,38 +1878,221 @@ static bool isFsIndexVar(ForallStmt* fs, Symbol* sym)
 // A forall-intents variation on findOuterVars() in createTaskFunctions.cpp:
 // Find all symbols used in 'block' and defined outside of it.
 //
-static void findOuterVarsNew(ForallStmt* fs, BlockStmt* block, SymbolMap& uses) {
+static void findOuterVarsNew(ForallStmt* fs, BlockStmt* block,
+                             FIIvec& fivec, FIImap& fimap)
+{
   std::vector<SymExpr*> symExprs;
 
   collectSymExprs(block, symExprs);
 
   for_vector(SymExpr, symExpr, symExprs) {
     Symbol* sym = symExpr->symbol();
 
-    if (isLcnSymbol(sym))
-      if (!isFsIndexVar(fs, sym) && isOuterVarNew(sym, block))
-        uses.put(sym, markUnspecified);
+    if (isLcnSymbol(sym) &&
+        // exclude a few things
```

Comment:
this is a useless comment. I can see that things are being excluded! It would be more useful to comment each line with the intent.  In particular I'd appreciate a comment next to isLcnSymbol saying "ArgSymbol or VarSymbol". Also, it seems a little strange to me that `this` is excluded. What's the justification for that? Can't `this` be an outer variable? Say, with a method containing an inner function? I can certainly see that gMethodToken can be ignored for this purpose.